module/channel/src/Infrastructure/Handler/DeleteChannelCommandHandler.php

```diff
@@ -10,27 +10,56 @@
 namespace Ergonode\Channel\Infrastructure\Handler;
 
 use Ergonode\Channel\Domain\Command\DeleteChannelCommand;
+use Ergonode\Channel\Domain\Query\ExportQueryInterface;
 use Ergonode\Channel\Domain\Repository\ChannelRepositoryInterface;
+use League\Flysystem\FilesystemInterface;
+use Psr\Log\LoggerInterface;
 use Webmozart\Assert\Assert;
 
 class DeleteChannelCommandHandler
 {
-    private ChannelRepositoryInterface $repository;
+    private ChannelRepositoryInterface $channelRepository;
 
-    public function __construct(ChannelRepositoryInterface $repository)
-    {
-        $this->repository = $repository;
+    private ExportQueryInterface $exportQuery;
+
+    private FilesystemInterface $exportStorage;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        ChannelRepositoryInterface $channelRepository,
+        ExportQueryInterface $exportQuery,
+        FilesystemInterface $exportStorage,
+        LoggerInterface $logger
+    ) {
+        $this->channelRepository = $channelRepository;
+        $this->exportQuery = $exportQuery;
+        $this->exportStorage = $exportStorage;
+        $this->logger = $logger;
     }
 
     /**
      * @throws \Exception
      */
     public function __invoke(DeleteChannelCommand $command): void
     {
-        $channel = $this->repository->load($command->getId());
+        $channel = $this->channelRepository->load($command->getId());
 
         Assert::notNull($channel, sprintf('Can\'t fid channel "%s"', $command->getId()->getValue()));
 
-        $this->repository->delete($channel);
+        $exportIds = $this->exportQuery->getExportIdsByChannelId($channel->getId());
+
+        $this->channelRepository->delete($channel);
+
+        foreach ($exportIds as $exportId) {
+            $file = sprintf('%s.zip', $exportId);
+            if ($this->exportStorage->has($file)) {
+                try {
+                    $this->exportStorage->delete($file);
```

Comment:
I guess there is no deletion of objects representing the export data together with the error list ?